tingbot/cache.py

```diff
@@ -0,0 +1,135 @@
+#image cache
+import time
+import re
+import rfc822
+import calendar
+import requests
+import io
+import os
+from urlparse import urlparse
+
+def get_http_timestamp(dt):
+    return calendar.timegm(rfc822.parsedate(dt))
```

Comment:
A few time comparisons in this file between different timebases (local vs. server) - could use the Date header in the HTTP response to correct the offset before storing.